javagen/src/main/java/com/azure/autorest/template/ServiceClientBuilderTemplate.java

```diff
@@ -516,17 +516,24 @@ protected void addCreateHttpPipelineMethod(JavaSettings settings, JavaClass clas
             function.line("policies.add(new CookiePolicy());");
 
             if (securityInfo.getSecurityTypes().contains(Scheme.SecuritySchemeType.KEY)) {
-                if (securityInfo.getHeaderName() == null
-                    || securityInfo.getHeaderName().isEmpty()) {
+                if (CoreUtils.isNullOrEmpty(securityInfo.getHeaderName())) {
                     logger.error("key-credential-header-name is required for " +
                             "azurekeycredential credential type");
                     throw new IllegalStateException("key-credential-header-name is required for " +
                             "azurekeycredential credential type");
                 }
                 function.ifBlock("azureKeyCredential != null", action -> {
-                    function.line("policies.add(new AzureKeyCredentialPolicy(\""
-                            + securityInfo.getHeaderName()
-                            + "\", azureKeyCredential));");
+                    if (CoreUtils.isNullOrEmpty(securityInfo.getHeaderValuePrefix())) {
```

Comment:
`AzureKeyCredentialPolicy` already support prefix? Have we tested it?

Reply:
added in 1.40 that just released, see description on test commit